plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -442,27 +467,32 @@ public String toString() {
 	
 	@Override
 	public <T extends IResource> List<T> getResources(String kind, String namespace) {
+		boolean needStrategy = initAuthorizationStrategy();
 		try {
-			if(client.getAuthorizationStrategy() == null) {
-				client.setAuthorizationStrategy(getAuthorizationStrategy());
-			}
-
 			client.setSSLCertificateCallback(OpenShiftCoreUIIntegration.getInstance().getSSLCertificateCallback());
 			return client.list(kind, namespace);
 		} catch (UnauthorizedException e) {
+			needStrategy = false;
 			return retryList("Unauthorized.  Trying to reauthenticate", e, kind, namespace);
+		} finally {
+			if(needStrategy) {
+				connect();
```

Comment:
The failed test is fixed in https://github.com/jbosstools/jbosstools-openshift/pull/1179